src/parquet/bloom.h

```diff
@@ -18,174 +18,140 @@
 #ifndef PARQUET_BLOOM_H
 #define PARQUET_BLOOM_H
 
-#include <set>
 #include <cstdint>
 
+#include "parquet/hasher.h"
 #include "parquet/types.h"
 #include "parquet/util/memory.h"
 
 namespace parquet {
 class OutputStream;
 
-// Bloom Filter is a compact structure to indicate whether an item is not in set or
-// probably in set. Bloom class is underlying class of Bloom Filter which stores a
-// bit set represents elements set, hash strategy and bloom filter algorithm.
+// A Bloom filter is a compact structure to indicate whether an item is not in set or
+// probably in set. Bloom class is underlying class of Bloom filter which stores a
+// bit set represents elements set, hash strategy and Bloom filter algorithm.
 
-// Bloom Filter algorithm is implemented using block Bloom filters from Putze et al.'s
-// "Cache-,Hash- and Space-Efficient Bloom Filters". The basic idea is to hash the
-// item to a tiny Bloom Filter which size fit a single cache line or smaller. This
-// implementation sets 8 bits in each tiny Bloom Filter. Tiny bloom filter are 32
+// Bloom filter algorithm is implemented using block Bloom filters from Putze et al.'s
+// "Cache-,Hash- and Space-Efficient Bloom filters". The basic idea is to hash the
+// item to a tiny Bloom filter which size fit a single cache line or smaller. This
+// implementation sets 8 bits in each tiny Bloom filter. Tiny Bloom filter are 32
 // bytes to take advantage of 32-bytes SIMD instruction.
 
 class Bloom final {
```

Comment:
I can add abstract for Algorithm too.

Do you suggest to respond to review (address comments) firstly and push a new commit to cover all comments?

Reply:
You state "the bloom algorithm abstract class is specific to Bloom filter, create a separated class seems too complex". I have to disagree. The idea of an interface to a filter, with methods like InsertHash and FindHash, seems not so complex. Furthermore, many possible implementation strategies exist for filters, including this block split hash, traditional bloom filters, cuckoo filters, quotient filters, and many more, and all of them support InsertHash and FindHash.

Reply:
OK, I will do this abstract in next commit.

Reply:
@jbapple-cloudera , I have updated code to use Bloom filter algorithm abstract class.  I didn't reply nits comments but I do update code for them since I think it would be too many noise for you.